app/driver/i2c_master.c

```diff
@@ -12,61 +12,50 @@
 #include "osapi.h"
 #include "gpio.h"
 #include "user_interface.h"
-
+#include "cpu_esp8266.h"
 #include "driver/i2c_master.h"
-
 #include "pin_map.h"
 
-LOCAL uint8 m_nLastSDA;
-LOCAL uint8 m_nLastSCL;
-
-LOCAL uint8 pinSDA = 2;
-LOCAL uint8 pinSCL = 15;
+#define CPU_CYCLES_BETWEEN_DELAYS 80
+#define MIN_SPEED 25000
+#define GPIO16_SCL_MAX_SPEED 400000
+#define GPIO16_RTC_REG_MASK 0xfffffffe
 
-LOCAL uint32 i2c_speed = 0;
-LOCAL int cycles_delay = 0;
+static i2c_master_state i2c[NUM_I2C];
```

Comment:
The easiest way to avoid this ram overhead is to have 
```C
static i2c_master_state *i2c;
static uint32 i2c_max;
```
and realloc on init if the current pin is larger than the current `i2C_max`.

But remember to check for NULL return on realloc

Reply:
Already done with static array of pointers to struct. Now it takes 40 bytes initially and +32 bytes for each bus after calling `i2c_master_gpio_setup'. I thought static array would be convenient to check if bus is initialized.

Reply:
:+1: